### DISCUSSION TITLE
Langium project for Theia extension.

### DISCUSSION THREAD

**Discussion by luxingxiao** [eclipse-langium/langium#1059] [Q&A] [answered]
When using langium's vscode plugin in Theia, there are many problems. Is there any plan to support Theia extension?

**Comment by spoenemann** [comment]
Could you describe in more detail what problems you are seeing? If there are issues with using a Langium-based language server in Theia, you should report them as bugs at the [theia repo](https://github.com/eclipse-theia/theia).

> Is there any plan to support Theia extension?

Note that Theia stopped supporting the direct integration of language servers since version 1.4.0 ([changelog](https://github.com/eclipse-theia/theia/blob/master/doc/changelogs/CHANGELOG-2020.md#1_4_0_deprecate_languages)). The only way to integrate Langium for text editor support is through a VS Code extension.

**Comment by msujew** [comment]
I've just tested the Langium vscode extension in Theia (`1.37.0`) and everything seems to work as expected. What issues are you seeing?

**Reply by luxingxiao** [reply]
@msujew Which example did you test?

**Reply by msujew** [reply]
I just installed the Langium extension itself from the open-vsx marketplace (version `1.2.0`). Given that we bootstrap the language (i.e. the grammar for Langium's grammar DSL is written in Langium), it should be enough to test whether Langium based language servers/vscode extensions run in Theia.

**Reply by luxingxiao** [reply]
![20230523-161845](https://github.com/langium/langium/assets/3615093/7a7abb03-eb9c-4947-8296-724b23a649f4)

@msujew  Left is vsocode, right is theia. Auto completion and error prompts features, obviously something is wrong.

**Reply by msujew** [reply]
Uhm, I'm not sure what the issue is there. I can't seem to reproduce this:

https://github.com/langium/langium/assets/4377073/d45054b9-287e-4126-99cc-2b66354dc9bf

**Reply by luxingxiao** [reply]
Is it because I am using the browser version? I've just tested it in theia blueprint online, the problems is still there.
![20230523-173640](https://github.com/langium/langium/assets/3615093/58a77ac5-c695-4d9d-9683-d9aa3898871b)

**Reply by msujew** [reply, marked as answer]
I get the same issue in the online blueprint version. It seems to be a serialization issue on the frontend side:

![image](https://github.com/langium/langium/assets/4377073/8c6d0f2c-9568-46da-8f7a-cc3ff71ebf8c)

It seems like it's related to a new version of the serialization library used. There's a fix in [Theia's migration guide](https://github.com/eclipse-theia/theia/blob/8ab9e3570c351376e76d33837bbde2a02a10e47b/doc/Migration.md#general). Can you try to apply that in your Theia project and see whether that works?

**Reply by luxingxiao** [reply]
It works! Thank you @msujew 👍

**Comment by luxingxiao** [comment]
Everything worked as expected in vscode, but some features often have problems in theia, such as auto completion and error prompts.

**Reply by luxingxiao** [reply]
Eclipse Theia: 1.37.0
Langium: 1.2.0

**Reply by msujew** [reply]
Do you have something specific or even a reproducible example to look at? These issues should be generally fixed on the Theia side though. I don't think there's really anything we can do on the level of Langium, given that we do nothing out of the ordinary.

Note that [I'm an active contributor to the Theia framework](https://github.com/eclipse-theia/theia/pulls?q=is%3Apr+author%3Amsujew), so I'm not trying to push the blame here. It would be good to see the issues in action so I could maybe contribute a fix.
